docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md

```diff
@@ -9,25 +9,245 @@ menu:
+
+## Examples
+
+This example uses follower reads since the **transaction** is marked read-only.
+
+```sql
+set yb_read_from_followers = true;
+start transaction read only;
+SELECT * from t WHERE k='k1';
+commit;
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+```
+
+This example uses follower reads since the **session** is marked read only.
+
+```sql
+set session characteristics as transaction read only;
+set yb_read_from_followers = true;
+SELECT * from t WHERE k='k1';
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+(1 row)
+```
+
+This next two examples use follower reads since **pg_hint** can be used during PREPARE and CREATE FUNCTION to do follower reads.
+
+```sql
+set session characteristics as transaction read write;
+set yb_read_from_followers = true;
+
+PREPARE select_stmt(text) AS
+/*+ Set(transaction_read_only on)*/
+SELECT * from t WHERE k=$1;
+
+EXECUTE select_stmt(‘k1’);
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+(1 row)
+```
+
+```sql
+set session characteristics as transaction read write;
+set yb_read_from_followers = true;
+
+CREATE FUNCTION func() RETURNS text AS
+$$ /*+ Set(transaction_read_only on)*/
+SELECT * from t WHERE k=1 $$ LANGUAGE SQL;
+CREATE FUNCTION
+SELECT func();
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+(1 row)
+```
+
+A join example that uses follower reads.
+
+```sql
+set yb_read_from_followers = true;
+set session characteristics as transaction read only;
+SELECT * from t1, t2 WHERE t1.k='k3' AND t1.v = t2.v;
+```
+
+```output
+ k | v
+---+---
+(0 rows)
+```
+
+The following examples demonstrate **staleness** after enabling the `yb_follower_read_staleness_ms` property.
+
+```sql
+set session characteristics as transaction read write;
+select * from t where k = 'k1';
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+(1 row)
+```
+
+```sql
+UPDATE t SET  v = 'v1+1' where k = 'k1';
+/* sleep 10s */
+
+UPDATE t SET  v = 'v1+2' where k = 'k1';
+/* sleep 10s */
+
+select * from t where k = 'k1';
+```
+
+```output
+ k  |  v
+----+------
+ k1 | v1+2
+(1 row)
+```
+
+```sql
+set session characteristics as transaction read only;
+set yb_read_from_followers = true;
+
+select * from t where k = 'k1';
+```
+
+```output
+ k  |  v
+----+------
+ k1 | v1
+(1 row)
+```
+
+```sql
+set yb_follower_read_staleness_ms = 5000;
+
+select * from t where k = 'k1';   /* 5s old value */
+```
+
+```output
+ k  |  v
+----+------
+ k1 | v1+2
+(1 row)
+```
+
+```sql
+set yb_follower_read_staleness_ms = 15000;
+
+select * from t where k = 'k1';   /* 15s old value */
+```
+
+```output
+ k  |  v
+----+------
+ k1 | v1+1
+(1 row)
+```
+
+```sql
+postgres=# set yb_follower_read_staleness_ms = 25000;
+postgres=# select * from t where k = 'k1';   /* 25s old value */
+```
+
+```output
+ k  |  v
+----+------
+ k1 | v1
+(1 row)
+```
+
+The final example demonstrates that SELECTS error out in read-only transactions.
```

Comment:
The wording needs to be improved here. 

It is not that SELECTs error out in a read-only transaction. It has more do with the fact that SELECT INTO is not a read-only transaction.

And this example has nothing to do with follower reads. Because we would get the same error even if follower reads was off.

So what's the point  of this example?

CC: @amitanandaiyer , @aishwarya24 ?

Reply:
Yes. That's true. We could just remove this example.

It was added to the FS to clarify the discussion regarding how we'd prevent/safegurard against a select statement  having non-read-only side-effects. 

Doesn't need to be part of the user documentation.